### PR TITLE
feat: bddtests - msgservice registration & handling using SDK

### LINKED ISSUE #1079: BDD tests : Messaging - Registration & Handling using SDK
From #974

### PR DESCRIPTION
**Pull request by sudeshrshetty**
- added messaging test feature in bddtests to demonstrate and test
message registrating registration and handling
- closes #1079

Signed-off-by: sudesh.shetty <sudesh.shetty@securekey.com>

### CODE REVIEW

File: test/bdd/pkg/context/context.go [hunk 1]
```diff
@@ -9,6 +9,10 @@ package context
 import (
 	"sync"
 
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/dispatcher"
+
```
Comment (troyronda) [inline test/bdd/pkg/context/context.go]: extra space

File: test/bdd/agent/agent_sdk_steps.go [hunk 2]
```diff
@@ -17,6 +17,8 @@ import (
 	"github.com/DATA-DOG/godog"
 	"github.com/google/uuid"
 
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/msghandler"
+
```
Comment (Moopli) [inline test/bdd/agent/agent_sdk_steps.go]: import order

File: test/bdd/pkg/context/context.go (hunk 1 above)
Comment (sudeshrshetty) [reply test/bdd/pkg/context/context.go]: fixed

File: test/bdd/pkg/messaging/msgservice.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+}
+
+func (m *msgService) Accept(header *service.Header) bool {
+	var purposeMatched, typeMatched = len(m.purpose) == 0, m.msgType == ""
+
+	if purposeMatched && typeMatched {
+		return false
+	}
+
+	for _, purposeCriteria := range m.purpose {
```
Comment (troyronda) [inline test/bdd/pkg/messaging/msgservice.go]: this code looks familiar?

File: test/bdd/agent/agent_sdk_steps.go (hunk 2 above)
Comment (sudeshrshetty) [reply test/bdd/agent/agent_sdk_steps.go]: fixed

File: test/bdd/pkg/messaging/msgservice.go (hunk 3 above)
Comment (sudeshrshetty) [reply test/bdd/pkg/messaging/msgservice.go]: this function is same as the one we use in rest api, I reused that logic for this test message service

**Pull request merged**